packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1055,6 +1056,24 @@ export class StripeHelper {
         continue;
       }
 
+      const result = subscriptionProductMetadataValidator.validate(
+        item.product.metadata
+      );
```

Comment:
This only validates the product's metadata.  See https://github.com/mozilla/fxa/pull/8437#pullrequestreview-657053420 about false negatives.  And see https://github.com/mozilla/fxa/pull/8437#issuecomment-840045090 on where to find an example of the metadata being merged.